contracts/Oracle.sol

```diff
@@ -3,10 +3,133 @@
+        Price memory batchPrice = batchPriceAt[_batch][_timestamp];
+        return (batchPrice.price, batchPrice.timestamp);
+    }
+
+    /**
+     * @notice get batch oracle
+     * @param _batch batch hash
+     * @return oracle address
+     */
+    function getBatchOracle(bytes32 _batch) public view returns (address) {
+        return batchOracle[_batch];
+    }
+
+    /**
+     * @notice get oracle locking period
+     * @dev during an oracle locking period, price can not be submitted to this contract
+     * @param _oracle oracle address
+     * @return locking period
+     */
+    function getOracleLockingPeriod(address _oracle) public view returns (uint256) {
+        return oracleLockingPeriod[_oracle];
+    }
+
+    /**
+     * @notice get oracle dispute period
+     * @dev during an oracle dispute period, the owner of this contract can dispute the submitted price and modify it
+     * @param _oracle oracle address
+     * @return dispute period
+     */
+    function getOracleDisputePeriod(address _oracle) public view returns (uint256) {
+        return oracleDisputePeriod[_oracle];
+    }
+
+    /**
+     * @notice check if locking period is over
+     * @param _batch batch hash
+     * @param _expiryTimestamp batch expiry
+     * @return True if locking period is over, otherwise false
+     */
+    function isLockingPeriodOver(bytes32 _batch, uint256 _expiryTimestamp) external view returns (bool) {
+        address oracle = batchOracle[_batch];
+        uint256 lockingPeriod = oracleLockingPeriod[oracle];
+
+        return now > _expiryTimestamp.add(lockingPeriod);
+    }
+
+    /**
+     * @notice check if dispute period is over
+     * @param _batch batch hash
+     * @param _expiryTimestamp batch expiry
+     * @return True if dispute period is over, otherwise false
+     */
+    function isDisputePeriodOver(bytes32 _batch, uint256 _expiryTimestamp) external view returns (bool) {
+        address oracle = batchOracle[_batch];
+        uint256 disputePeriod = oracleDisputePeriod[oracle];
+
+        Price memory batchPrice = batchPriceAt[_batch][_expiryTimestamp];
+
+        if (batchPrice.timestamp == 0) {
+            return false;
+        }
+
+        return now > batchPrice.timestamp.add(disputePeriod);
+    }
+
+    /**
+     * @notice set batch oracle
+     * @dev can only be called by owner
+     * @param _batch batch (hash of underlying, stike, collateral and expiry)
+     * @param _oracle oracle address
+     */
+    function setBatchOracle(bytes32 _batch, address _oracle) external onlyOwner {
+        batchOracle[_batch] = _oracle;
+    }
+
+    /**
+     * @notice set oracle locking period
+     * @dev can only be called by owner
+     * @param _oracle oracle address
+     * @param _lockingPeriod locking period
+     */
+    function setLockingPeriod(address _oracle, uint256 _lockingPeriod) external onlyOwner {
+        oracleLockingPeriod[_oracle] = _lockingPeriod;
```

Comment:
Should we have events for this and `setDisputePeriod`?

Reply:
yeah will add events in the setter PR.